### ISSUE TITLE
Duplication between PublicAPI.Shipped.txt and PublicAPI.Unshipped.txt with nullable attributes

### ISSUE THREAD

**Issue by JamesNK** [dotnet/roslyn-analyzers#4584]
PublicAPI.Shipped.txt:

```
#nullable enable
Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.get -> int
Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.set -> void
```

PublicAPI.Unshipped.txt:

```
#nullable enable
Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.get -> int
Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.set -> void
```

Analyzer correctly reports an error:

```
1>C:\Development\Source\AspNetCore\src\Middleware\Diagnostics\src\PublicAPI.Unshipped.txt(2,1,2,90): error RS0025: The symbol 'Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.get -> int' appears more than once in the public API files.
1>C:\Development\Source\AspNetCore\src\Middleware\Diagnostics\src\PublicAPI.Unshipped.txt(3,1,3,91): error RS0025: The symbol 'Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.set -> void' appears more than once in the public API files.
```

But if there is ~ present on APIs in one of the files, then the duplication is not detected. PublicAPI.Unshipped.txt:

```
#nullable enable
~Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.get -> int
~Microsoft.AspNetCore.Builder.DeveloperExceptionPageOptions.SourceCodeLineCount.set -> void
```

The warning should still happen to prevent duplication (which could happen because of a bad merge, etc)
